server/compose/types/record_detect_duplicates.go

```diff
@@ -146,6 +156,30 @@ func (rule DeDupRule) checkCaseSensitiveDuplication(ctx context.Context, ls loca
 			valErr = &RecordValueErrorSet{}
 		)
 
+		existingVv := vv.FilterByName(c.Attribute)
+		if c.IsAllEqual() {
+			if rvv.Len() == existingVv.Len() {
+				rvvmap := recordValueFrequencyMap(c.Modifier, rvv)
+				existingVvmap := recordValueFrequencyMap(c.Modifier, existingVv)
```

Comment:
Might be worth considering memory pressure; since this would have a very tight loop, it might impact performance due to heavy allocations cause of `map`.

It might all get inlined so it might not make sense so should be checked